src/test/java/seedu/address/logic/commands/DeleteCommandTest.java

```diff
@@ -66,26 +88,51 @@ public void execute_validIndexFilteredList_success() {
     @Test
     public void execute_invalidIndexFilteredList_throwsCommandException() {
         showAssignmentAtIndex(model, INDEX_FIRST_ASSIGNMENT);
-
+        List<Index> assignmentIndexesToDelete = new ArrayList<>();
         Index outOfBoundIndex = INDEX_SECOND_ASSIGNMENT;
+        assignmentIndexesToDelete.add(outOfBoundIndex);
         // ensures that outOfBoundIndex is still in bounds of address book list
         assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getAssignmentList().size());
 
-        DeleteCommand deleteCommand = new DeleteCommand(outOfBoundIndex);
+        DeleteCommand deleteCommand = new DeleteCommand(assignmentIndexesToDelete);
 
         assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
     }
 
+    @Test
+    public void execute_duplicatedIndexFilteredList_throwsCommandException() {
+        showAssignmentAtIndex(model, INDEX_FIRST_ASSIGNMENT);
+        List<Index> assignmentIndexesToDelete = new ArrayList<>();
+        Index duplicatedIndex = INDEX_FIRST_ASSIGNMENT;
+        assignmentIndexesToDelete.add(duplicatedIndex);
+        assignmentIndexesToDelete.add(duplicatedIndex);
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(duplicatedIndex.getZeroBased() < model.getAddressBook().getAssignmentList().size());
+
+        DeleteCommand deleteCommand = new DeleteCommand(assignmentIndexesToDelete);
+
+        assertCommandFailure(deleteCommand, model, String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand
+                .MESSAGE_ASSIGNMENTS_DUPLICATED_INDEX));
+    }
+
+
     @Test
     public void equals() {
-        DeleteCommand deleteFirstCommand = new DeleteCommand(INDEX_FIRST_ASSIGNMENT);
-        DeleteCommand deleteSecondCommand = new DeleteCommand(INDEX_SECOND_ASSIGNMENT);
+        List<Index> firstCommandIndexes = new ArrayList<>();
+        firstCommandIndexes.add(INDEX_FIRST_ASSIGNMENT);
+        firstCommandIndexes.add(INDEX_THIRD_ASSIGNMENT);
```

Comment:
Maybe adding comments to explain what these are would be good as I am currently a bit lost as to why INDEX_FIRST_ASSIGNMENT was added twice :)